reference/shaders-hlsl/frag/basic.frag

```diff
@@ -0,0 +1,31 @@
+uniform sampler2D uTex;
```

Comment:
I recall stuff like:
```
Texture2D<float4> tex1[4] : register(t3)
```
Also, are layout()s handled in any way, or is it up to the application to remap before compiling the HLSL through some API?